tartufo/util.py

```diff
@@ -57,10 +58,17 @@ def clean_outputs(output_dir: pathlib.Path) -> None:
         shutil.rmtree(output_dir)
 
 
-def clone_git_repo(git_url: str) -> str:
-    project_path = tempfile.mkdtemp()
+def clone_git_repo(
+    git_url: str, target_dir: Optional[pathlib.Path] = None
+) -> pathlib.Path:
+    if not target_dir:
+        project_path = tempfile.mkdtemp()
+    else:
+        repo_name = urlparse(git_url).path.split("/")[-1]
+        project_path = str(target_dir / repo_name)
```

Comment:
`target_dir` would imply the code would get cloned into that directory. If I said `git clone github.com/myorg/myrepo /tmp/my_target_dir` I would see the code there and not `/tmp/my_target_dir/myrepo`. I would have the caller append repo_name.

Reply:
I had that implementation originally and found that it lead to potentially confusing behavior due to the fact that we remove the repo after it's scanned. So let's say I ran:

`tartufo scan-remote-repo --work-dir=~/temp https://github.com/godaddy/tartufo.git`

It would clone the repository (without a containing folder) directly into `~/temp` and scan it there. It would then remove the entire `~/temp` folder, which is NOT what I would expect as an end user.

It felt to me far more reasonable to clone into `~/temp/tartufo.git/` and then remove that `tartufo.git` folder afterward, leaving the target directory in tact, the way it was before hand.

Reply:
Correct. You are referring to `work-dir`. I agree. I'm referring to this method. `clone` and `target_dir` I would imagine would operate similar to `git clone` with target dir.

Reply:
Ah. So you are proposing to move this `target_dir / repo_name` bit to be moved into the `scan-remote-repo` command then? I think that makes sense. I'll just have to make sure this method gets documented to make that absolutely clear.

Reply:
@dclayton-godaddy There ya go, I shifted the path manipulation up to the command itself. Should cover your concerns. Good call on that.